indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -962,123 +340,83 @@ class RoomEditModal extends React.Component {
                     size="mini"
                     color="red"
                     content={Translate.string('Reset form')}
-                    onClick={() => {
-                      // reset the form
-                      this.setState({submitState: ''});
-                      fprops.form.reset();
-                    }}
+                    onClick={() => formProps.form.reset()}
                   />
                 </Message>
+              );
+            }}
+          </FormSpy>
+          <Grid columns="equal">
+            {!isNewRoom && (
+              <Grid.Column width={4}>
+                <RoomPhoto roomId={roomId} hasPhoto={roomDetails && roomDetails.has_photo} />
               </Grid.Column>
-            </Grid>
-          </Form>
+            )}
+            <Grid.Column>
+              <Form id="room-form" onSubmit={handleFormSubmit}>
+                <Tab renderActiveOnly={false} panes={tabPanes} />
+              </Form>
+            </Grid.Column>
+          </Grid>
         </Modal.Content>
         <Modal.Actions>
-          <Button onClick={this.handleCloseModal}>
-            {submitState === 'success' ? (
-              <Translate>Close</Translate>
-            ) : (
-              <Translate>Cancel</Translate>
+          <FormSpy subscription={{submitSucceeded: true}}>
+            {({submitSucceeded}) => (
+              <Button onClick={onClose}>
+                {submitSucceeded ? <Translate>Close</Translate> : <Translate>Cancel</Translate>}
```

Comment:
Not sure if that does what you intend: The old `submitState` was reset as soon as something changed in the form (so it went back from Close to Cancel if you started editing after saving) - I don't think this happens with `submitSucceeded`.

Reply:
Yes, the only difference I saw was not resetting after a successful submission. 
However, do you think it is worth changing this to the internal state (another re-render) just because of a button change after the first submit?
It could also seem reasonable for the button to say `Close` if you already saved previously.